Jenkinsfile

```diff
@@ -0,0 +1,61 @@
+pipeline {
+    agent none
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'5'))
+    }
+    stages {
+        stage('Setup') {
+            steps{
+                slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+            }
+        }
+        stage('Parallel Build') {
+            // TODO refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
+            steps{
+                parallel(
+                    linux: {
+                        node('linux-large') {
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings') {
+                                    sh 'mvn clean install -pl !test-itest-ddf'
+                                    sh 'mvn install -pl test-itest-ddf'
+                                }
+                            }
+                        }
+                    }, windows: {
+                        node('proxmox-windows'){
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings') {
+                                    bat 'mvn clean install -pl !test-itest-ddf'
+                                    bat 'mvn install -pl test-itest-ddf'
+                                }
+                            }
+                        }
+                    }
+                )
+            }
+        }
+        stage('Deploy') {
+            agent { label 'linux-small' }
+            steps{
+                withMaven(maven: 'M3', globalMavenSettingsConfig: 'phxOfficeGlobalSettings', mavenSettingsConfig: 'codice_settings.xml') {
+                    sh 'mvn javadoc:aggregate -Pjavadoc -DskipStatic=true -DskipTests=true'
+                    sh 'mvn deploy -DskipStatic=true -DskipTests=true'
```

Comment:
Initially this jenkinsfile will only serve the purpose of a "Nightly" build and as such does not take into account the flow control that will be necessary (and added) as part of the follow on work. The expectation here is that the javadoc plus the artifacts will be uploaded with these two scripts.